Add an entry on Lightning channel jamming

s-tikhomirov opened the pull request:
An overview of jamming and the proposed approaches to addressing it. Corresponding issue: #30.

LLFourn commented on _problems/channel-jamming.md:
I imagine the proposed solutions section to include references to works whose authors confidently believe they have a solution or at least partial solution that is worked out enough that it could be implemented. I think they should always include a link to the solution. e.g.

> In *[Mitigating Channel Jamming with Stake Certificates](https://thelab31.xyz/blog/stake-certificates)* Naumenko and Riard propose.... 

and then include  short summary of the proposal.

However I'm not sure if all of the things here are proposed solutions -- e.g. stake certificates isn't a completed idea. It should probably under a heading in the potential directions section.

Even for those ideas that become "Potential Directions" more inline links to any related discussion would still be helpful.

s-tikhomirov replied:
> I'm not sure if all of the things here are proposed solutions

I agree. Actually

LLFourn commented on _problems/channel-jamming.md:
Since it's not a solution perhaps it could under a heading `Practical Mitigations` in the description.

s-tikhomirov replied:
> under a heading Practical Mitigations in the description

Do you see `Practical Mitigation` as another first-level (`##`) heading?

LLFourn reviewed (changes requested):
Fantastic and important work.  With a few more inline links in the Proposed Solutions section it'll be a very useful resource. There are some important ideas I haven't heard before in here but I'm not sure if they're your original research or they come from somewhere else.

Since @t-bast is already maintaining https://github.com/t-bast/lightning-docs/blob/master/spam-prevention.md#proposals I wonder if we could merge the efforts somehow and transfer your additions into that document and just link to it?

t-bast commented:
> Since @t-bast is already maintaining https://github.com/t-bast/lightning-docs/blob/master/spam-prevention.md#proposals I wonder if we could merge the efforts somehow and transfer your additions into that document and just link to it?

Sure, don't hesitate to open a PR to https://github.com/t-bast/lightning-docs if you feel like it, and I'll spend some time reviewing it. But if you'd rather keep this in the `bitcoin-problems` repo, that's also fine for me.

s-tikhomirov commented:
> don't hesitate to open a PR to https://github.com/t-bast/lightning-docs if you feel like it, and I'll spend some time reviewing it

Thanks @t-bast ! I think though it makes sense to keep these two documents separate. I see your gist a more implementation-focused, while the bitcoin-problems entry should arguably be more high-level and targeted at researchers who may be unfamiliar with fine-grained details of the protocol.

s-tikhomirov commented:
@LLFourn : I've done some more editing that I believe addresses your feedback, re-requesting review.

Regarding the directions vs proposals distinction: I agree that these are different things, although the line is blurry. Strictly speaking, I wouldn't call any of the ideas an implementation-ready proposal. The upfront fees protocols seem to be the closest to this status, so I left that discussion under the "solutions" heading. Configuration changes now are under a new "mitigations" heading, and everything else is under "research directions". Do you think this makes sense?

LLFourn reviewed (approved):
Thanks! I really like the way this is structured. The problems and potential solutions are very clear. Fantastic work.
